File: src/main/java/io/github/kingg22/api/vacunas/panama/service/FabricanteService.java
package io.github.kingg22.api.vacunas.panama.service;

import io.github.kingg22.api.vacunas.panama.modules.fabricante.entity.Fabricante;
import io.github.kingg22.api.vacunas.panama.modules.fabricante.repository.FabricanteRepository;
import io.github.kingg22.api.vacunas.panama.modules.fabricante.service.IFabricanteService;
import jakarta.validation.constraints.NotNull;
import java.util.Optional;
import java.util.UUID;
import org.springframework.stereotype.Service;

/** Service for {@link Fabricante}. */
@Service
public class FabricanteService implements IFabricanteService {
    private final FabricanteRepository fabricanteRepository;

    public FabricanteService(FabricanteRepository fabricanteRepository) {
        this.fabricanteRepository = fabricanteRepository;
    }

    @org.jetbrains.annotations.NotNull
    public Optional<Fabricante> getFabricante(@org.jetbrains.annotations.NotNull @NotNull String licencia) {
        return fabricanteRepository.findByLicencia(licencia);
    }

    @org.jetbrains.annotations.NotNull
    public Optional<Fabricante> getFabricanteByUserID(@org.jetbrains.annotations.NotNull @NotNull UUID idUser) {
        return this.fabricanteRepository.findByUsuario_Id(idUser);
    }
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/modules/fabricante/service/FabricanteService.kt
package io.github.kingg22.api.vacunas.panama.modules.fabricante.service

import io.github.kingg22.api.vacunas.panama.modules.fabricante.dto.FabricanteDto
import java.util.UUID

/** Service interface for managing and retrieving fabricante-related information. */
interface FabricanteService {
    /**
     * Retrieves a [FabricanteDto] associated with the specific licencia.
     *
     * @param licencia String with the licencia linked to the fabricante.
     * @return The fabricante if found or null if not.
     */
    fun getFabricante(licencia: String): FabricanteDto?

    /**
     * Retrieves a [FabricanteDto] associated with the specified user ID.
     *
     * @param idUser UUID of the user linked to the doctor.
     * @return The fabricante if found or null if not.
     */
    fun getFabricanteByUserID(idUser: UUID): FabricanteDto?
}
